utils/validationSchemas.ts

```diff
@@ -0,0 +1,46 @@
+import { object, string, number, date, InferType, ref } from "yup";
+
+import { isValidPhoneNumber } from "libphonenumber-js";
+
+// please move all message in `constants/messages.ts` file
+const requiredFieldMessage = "Ce champs est requis";
+const mustBeValidPhoneNumber = "Ajoutez un numéro de téléphone valide";
+const mustBeAEmailAdress = "Ajoutez une adresse mail valide";
+const passwordRequirementMsg =
+  "Le mot de passe doit contenir au moins 8 caractères, une lettre majuscule, une lettre minuscule, un chiffre et l'un des caractères spéciaux suivants @$!%*?&.";
+
+const passwordRegex =
+  "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]{8,}$";
+
+const phoneNumberValidator = string()
+  .required(requiredFieldMessage)
+  .test(
+    "Ajoutez un numéro de téléphone valide",
+    (value) => !!(value && isValidPhoneNumber(value)),
+  );
+
+export const loginValidationSchema = object({
+  email: string().required(requiredFieldMessage).email(mustBeAEmailAdress),
+  password: string()
+    .required(requiredFieldMessage)
+    .matches(new RegExp(passwordRegex), passwordRequirementMsg),
+  createdOn: date().default(() => new Date()),
```

Comment:
Why this ?
it's ChatGPT 🫣 ?

Reply:
Non, je pensais que cette ligne permet de garder la date de l'instance de création de compte/ login

Reply:
Je vois. Cette date sera définie autrement lors l'appel de Firebase pour la sauvegarde. Ici, c'est juste pour la validation.